### PR TITLE
chore(Alert): updated tests to new standards

### LINKED ISSUE #7670: Alert test revamp

### PR DESCRIPTION
**Pull request by wise-king-sullyman**
**What**: Closes #7670 

**Additional issues**:

I also resolved a few small bugs in 749bf18 which I discovered in `AlertActionCloseButton` and `AlertToggleExpandButton` while building out the test suites.

### CODE REVIEW

File: packages/react-core/src/components/Alert/__tests__/AlertIcon.test.tsx [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+import { AlertIcon } from '../AlertIcon';
+
+jest.mock('@patternfly/react-icons/dist/esm/icons/check-circle-icon', () => () => 'Check circle icon mock');
+jest.mock('@patternfly/react-icons/dist/esm/icons/exclamation-circle-icon', () => () => 'Exclamation circle icon mock');
+jest.mock('@patternfly/react-icons/dist/esm/icons/exclamation-triangle-icon', () => () =>
+  'Exclamation triangle icon mock'
+);
+jest.mock('@patternfly/react-icons/dist/esm/icons/info-circle-icon', () => () => 'Info circle icon mock');
+jest.mock('@patternfly/react-icons/dist/esm/icons/bell-icon', () => () => 'Bell icon mock');
```
Comment (nicolethoen) [inline packages/react-core/src/components/Alert/__tests__/AlertIcon.test.tsx:12]: 🤯 good to know this is how it works!

**Review by nicolethoen — COMMENTED** [review]
so, in the back of my mind I'm aware that right now, the a11y tests to flag a warning that we are putting an aria-label on a div without a valid role. And in the case of alert, the aria-label is the same text as the title of the alert (by default at least). So it'd be considered a breaking change at this point to remove the default aria-label, because of the unit tests in product, and now these unit tests.

I'm not sure what that means for this approach. When/if we implement a fix to avoid the warning, then we'll also need to update these unit tests and I guess that's okay. I just wanted to document the thought process i guess. 

And yeah, outside of that, maybe we can make the timeouts shorter whenever possible - like instead of 3000 milliseconds, use 100 milliseconds or something like that?

File: packages/react-core/src/components/Alert/__tests__/Alert.test.tsx [hunk 2]
```diff
@@ -1,210 +1,724 @@
...
+      Some alert
+    </Alert>
+  );
+
+  const alert = screen.getByText('Some alert');
+
+  await user.hover(alert);
+
+  act(() => {
+    jest.advanceTimersByTime(8000);
```
[the remark this answers is not in the document]
Comment (wise-king-sullyman) [reply packages/react-core/src/components/Alert/__tests__/Alert.test.tsx:416]: I could be mistaken, but I believe that Jest's fake timer functionality here actually allows the timers to advance in non-real time, so the 8000ms here should actually be passing pretty much instantly.

File: packages/react-core/src/components/Alert/__tests__/AlertToggleExpandButton.test.tsx [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+  const onCloseMock = jest.fn();
+
+  render(
+    <AlertContext.Provider value={{ title: 'title', variantLabel: 'variantLabel' }}>
+      <AlertToggleExpandButton onToggleExpand={onCloseMock} />
+    </AlertContext.Provider>
+  );
+
+  expect(onCloseMock).not.toHaveBeenCalled();
+});
```
Comment (thatblindgeye) [inline packages/react-core/src/components/Alert/__tests__/AlertToggleExpandButton.test.tsx:65]: Nit: could we update this test to reflect the onToggleExpand is being tested instead of onClose?

File: packages/react-core/src/components/Alert/__tests__/AlertToggleExpandButton.test.tsx [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+  render(
+    <AlertContext.Provider value={{ title: 'title', variantLabel: 'variantLabel' }}>
+      <AlertToggleExpandButton onToggleExpand={onCloseMock} />
+    </AlertContext.Provider>
+  );
+
+  await user.click(screen.getByRole('button'));
+
+  expect(onCloseMock).toHaveBeenCalledTimes(1);
+});
```
Comment (thatblindgeye) [inline packages/react-core/src/components/Alert/__tests__/AlertToggleExpandButton.test.tsx:80]: Same thing here regarding the above comment.

File: packages/react-core/src/components/Alert/__tests__/Alert.test.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (nicolethoen) [reply packages/react-core/src/components/Alert/__tests__/Alert.test.tsx:416]: ooooo that's cool!

**Review by nicolethoen — APPROVED** [review]

File: packages/react-core/src/components/Alert/__tests__/Alert.test.tsx [hunk 5]
```diff
@@ -1,210 +1,724 @@
...
+  const alert = screen.getByText('Some alert');
+
+  await user.click(alert);
+
+  act(() => {
+    jest.advanceTimersByTime(8000);
+  });
+
+  await user.click(screen.getByRole('textbox'));
+
```
Comment (thatblindgeye) [inline packages/react-core/src/components/Alert/__tests__/Alert.test.tsx:447]: For this and the following test, would it make sense to check that the alert is still there after 8000ms and before clicking/hovering away from the alert? Or would the previous tests that cover that be enough?

File: packages/react-core/src/components/Alert/__tests__/Alert.test.tsx [hunk 6]
```diff
@@ -1,210 +1,724 @@
...
+  const alert = screen.getByText('Some alert');
+
+  await user.hover(alert);
+
+  act(() => {
+    jest.advanceTimersByTime(8000);
+  });
+
+  await user.hover(screen.getByRole('textbox'));
```
Comment (thatblindgeye) [inline packages/react-core/src/components/Alert/__tests__/Alert.test.tsx:479]: Same question as the above comment.

File: packages/react-core/src/components/Alert/__tests__/Alert.test.tsx [hunk 7]
```diff
@@ -1,210 +1,724 @@
...
```
Comment (thatblindgeye) [inline packages/react-core/src/components/Alert/__tests__/Alert.test.tsx:512]: Didn't realize there was another similar test, but same question re: testing if the alert still exists after 8000ms and before moving hover.

**Review by thatblindgeye — COMMENTED** [review]
This looks great, awesome work on all of this! I had a couple of changes below (should be the first two comments), then a few comments asking the same question. Let me know what you think!

File: packages/react-core/src/components/Alert/__tests__/AlertToggleExpandButton.test.tsx (hunk 3 above)
Comment (wise-king-sullyman) [reply packages/react-core/src/components/Alert/__tests__/AlertToggleExpandButton.test.tsx:65]: Yes! Great catch!

File: packages/react-core/src/components/Alert/__tests__/Alert.test.tsx (hunk 5 above)
Comment (wise-king-sullyman) [reply packages/react-core/src/components/Alert/__tests__/Alert.test.tsx:447]: Yeah I debated that myself, but elected to not have that extra check in this test since that aspect of the behavior is already being covered in the preceding test. Happy to add it in if you don't think the other test is enough coverage of it though.

Comment (thatblindgeye) [reply packages/react-core/src/components/Alert/__tests__/Alert.test.tsx:447]: I think the only scenario I can imagine where it might come up is if hovering/focusing an alert no longer prevented dismissal (mainly if something else were added/edited/removed to somehow break that functionality). In which case I believe the test would still pass, but the hover/click wouldn't have prevented the dismissal?

That might be quite the edgy edge case, though, so I'd be fine with these three tests as-is if you feel the same way.

**Review by thatblindgeye — APPROVED** [review]
🎉 💪🏼
